functions/knative/java/cloudevent/README.md

```diff
@@ -0,0 +1,149 @@
+# Sample Function Java
+
+## Prerequisites
+
+### OpenFunction
+
+You can refer to the [Installation Guide](https://openfunction.dev/docs/getting-started/installation/) to setup OpenFunction.
+
+## Run it locally
+
+Build the function locally
+
+  ```sh
+  pack build func-cloudevent-java --path ../ --builder openfunction/builder-java:v2-18 --env FUNC_NAME="dev.openfunction.samples.CloudEventFunctionImpl"  --env FUNC_CLEAR_SOURCE=true
+  ```
+
+Run the function
+
+  ```sh
+  docker run --rm --env="FUNC_CONTEXT={\"name\":\"CloudEvent\",\"version\":\"v1.0.0\",\"port\":\"8080\",\"runtime\":\"Knative\"}" --env="CONTEXT_MODE=self-host" --name func-cloudevent-java -p 8080:8080 func-cloudevent-java
+  ```
+
+Send a request
+
+  ```sh
+  curl -X POST "http://localhost:8080/" \
+     -H "Content-Type: application/cloudevents+json" \
+     -d '{"specversion":"1.0","type":"dev.knative.samples.helloworld","source":"dev.knative.samples/helloworldsource","id":"536808d3-88be-4077-9d7a-a3f162705f79","data":"{\"data\":\"hello world\"}"}'
+  # in docker compose terminal:
+  # receive event: {"data":"hello world"}
+
+  # Binary CloudEvent
+  curl "http://localhost:8080/" \
+    -X POST \
+    -H "Ce-Specversion: 1.0" \
+    -H "Ce-Type: dev.knative.samples.helloworld" \
+    -H "Ce-Source: dev.knative.samples/helloworldsource" \
+    -H "Ce-Subject: 123" \
+    -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+    -H "Content-Type: application/json" \
+    -d '{"data":"hello world"}'
+  # in docker compose terminal:
+  # receive event: {"data":"hello world"}
+  ```
+
+## Deployment
+
+1. Create secret
+
+Follow [this guide](../../../../Prerequisites.md#registry-credential) to create a registry credential.
```

Comment:
Use the prereq guide in : https://openfunction.dev/docs/getting-started/quickstarts/prerequisites/